### PR TITLE
Read dune files in addition to jbuild files

### LINKED ISSUE #81: Doesn't find dune files
On a native project that builds with dune the language server reports that it is unable to find a jbuild file. We should probably be looking for dune files first, falling back to jbuild files, then failing.

### PR DESCRIPTION
**Pull request by zploskey**
Fixes #81.

### CODE REVIEW

File: src/analyze/State.re [hunk 1]
```diff
@@ -169,7 +169,7 @@ let newJbuilderPackage = (rootPath) => {
   let (otherDirectories, otherFiles) = source |> List.filter(s => s != "." && s != "" && s.[0] == '.') |> optMap(name => {
     let otherPath = rootPath /+ name;
     let res = {
-      let%try jbuildRaw = Files.readFileResult(otherPath /+ "jbuild");
+      let%try jbuildRaw = JbuildFile.readFromDir(otherPath /+ "jbuild");
```
Comment (jaredly) [inline src/analyze/State.re]: `otherPath /+ "jbuild"` -> `otherPath/`

Comment (zploskey) [reply src/analyze/State.re]: Good catch, copy-paste error. Fixed and squashed.

**Comment by zploskey** [comment]
@jaredly I think this is done. The example dune project works with these changes.

**Comment by zploskey** [comment]
The launcher tries to run `npm install` on the project which won't work for an esy project. Not sure how to change that off-hand.

**Comment by jaredly** [comment]
Awesome, thanks for fixing!

**Pull request merged**